src/sync/outgoingSyncUtils.js

```diff
@@ -233,3 +258,21 @@ function getTimeString(date) {
   if (!date || typeof date !== 'object') return '00:00:00';
   return date.toTimeString().substring(0, 8);
 }
+
+// Tries to get a value that is known to potentially lead to crash
+// If path on the record returns null, throw an error with prototype
+// extended with 'canDeleteSyncOut' set to true so sync knows to continue
+function safeGet(record, path) {
+  const pathSegments = path.split('.');
+  let nestedProp = record;
+  for (let i = 0; i < pathSegments.length; i++) {
```

Comment:
i was thinking more function safeGet(record, ...paths) can then call it safeGet (record, itemBatch, id) rather then safeGet(record, 'itemBatch.id').
Can then iterate over paths, and if error can paths.join('.') to pass more info to bug snag (by the way that's missing now)

Reply:
I like the single string as that path arg as it more closely resembles what you'd be doing if you were just doing dot notation on an object.

Will make the error better as you suggest though.